src/tree/StaticWebAppTreeItem.ts

```diff
@@ -62,7 +62,15 @@ export class StaticWebAppTreeItem extends AzureParentTreeItem implements IAzureR
             envs,
             'invalidStaticEnvironment',
             async (env: WebSiteManagementModels.StaticSiteBuildARMResource) => {
-                return await EnvironmentTreeItem.createEnvironmentTreeItem(context, this, env);
+                try {
+                    return await EnvironmentTreeItem.createEnvironmentTreeItem(context, this, env);
+                } catch (error) {
+                    const parsedError: IParsedError = parseError(error);
+                    if (/(null|undefined).*sourceBranch/.test(parsedError.message)) {
```

Comment:
`sourceBranch` (which gets assigned to `EnvironmentTreeItem.branch`) isn't actually used that often.  With the first scenario, what functionality do we offer to users without having the branch defined?  

Most of our features are tied to GitHub, but we might still be able to offer them basic commands such as open in portal, browse, view properties.  It might be better to support this minimally.

Reply:
I don't disagree, but can't hurt to have a slightly better error message first. Internal errors are never supposed to be shown to the user